lib/clientv2/workloadendpoint_e2e_test.go

```diff
@@ -397,4 +410,106 @@ var _ = testutils.E2eDatastoreDescribe("WorkloadEndpoint tests", testutils.Datas
+					Spec:       apiv2.WorkloadEndpointSpec{
+						Node: "node-1",
+						Orchestrator: "k8s",
+						Pod: "pod-1",
+						Endpoint: "eth0",
+						InterfaceName: "cali1234",
+					},
+				},
+				options.SetOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating a workload in a different namespace, but with a largely overlapping name")
+			outRes3, err := c.WorkloadEndpoints().Create(
+				ctx,
+				&apiv2.WorkloadEndpoint{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "namespace2", Name: "node--1-k8s-pod--2-eth0"},
+					Spec:       apiv2.WorkloadEndpointSpec{
+						Node: "node-1",
+						Orchestrator: "k8s",
+						Pod: "pod-2",
+						Endpoint: "eth0",
+						InterfaceName: "cali1235",
+					},
+				},
+				options.SetOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Doing an exact get on one of the workload endpoints")
+			outList, err := c.WorkloadEndpoints().List(ctx, options.ListOptions{Namespace: "namespace1", Name: "node--1-k8s-pod-eth0"})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(outList.Items).To(HaveLen(1))
+			testutils.ExpectResource(&outList.Items[0], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod-eth0", outRes1.Spec)
+
+			By("Doing a short prefix get to retrieve both workload endpoints in namespace1")
+			outList, err = c.WorkloadEndpoints().List(ctx, options.ListOptions{Namespace: "namespace1", Name: "node--1-k8s-pod-", Prefix: true})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(outList.Items).To(HaveLen(2))
+			testutils.ExpectResource(&outList.Items[0], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod--1-eth0", outRes2.Spec)
+			testutils.ExpectResource(&outList.Items[1], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod-eth0", outRes1.Spec)
+
+			By("Doing a longer prefix get to retrieve one workload endpoints in namespace1")
+			outList, err = c.WorkloadEndpoints().List(ctx, options.ListOptions{Namespace: "namespace1", Name: "node--1-k8s-pod--1", Prefix: true})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(outList.Items).To(HaveLen(1))
+			testutils.ExpectResource(&outList.Items[0], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod--1-eth0", outRes2.Spec)
+
+			By("Doing a short prefix get with wildcarded namespace to retrieve all workload endpoints")
+			outList, err = c.WorkloadEndpoints().List(ctx, options.ListOptions{Name: "node--1-k8s-pod-", Prefix: true})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(outList.Items).To(HaveLen(3))
+			testutils.ExpectResource(&outList.Items[0], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod--1-eth0", outRes2.Spec)
+			testutils.ExpectResource(&outList.Items[1], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod-eth0", outRes1.Spec)
+			testutils.ExpectResource(&outList.Items[2], apiv2.KindWorkloadEndpoint, "namespace2", "node--1-k8s-pod--2-eth0", outRes3.Spec)
+
+			By("Doing a long prefix get with wildcarded names to retrieve the workload endpoint in namespace2")
```

Comment:
not sure what this test is buying us. Not too fussed about it tho

Reply:
it's just testing that the filtering is working correctly when namespace is wild carded.